src/app/data-model/List.h

```diff
@@ -72,6 +78,21 @@ inline CHIP_ERROR Encode(TLV::TLVWriter & writer, TLV::Tag tag, List<X> list)
     return CHIP_NO_ERROR;
 }
 
+template <typename X, std::enable_if_t<DataModel::IsFabricScoped<X>::value, int> = 0>
```

Comment:
Should condition the Encode() method on `X` not being fabric-scoped, I assume, right?

And we probably do want an `EncodeForRead` here for cases when someone encodes a list as a single thing, without chunking....

Reply:
I think by virtue of the Encode automatically failing given the absence of `EncodeForRead/Write` APIs on the attendant cluster objects should be sufficient for now.

Will add the `EncodeForRead`.